### PR TITLE
Call uniq! on the `reorder` arguments.

### PR DESCRIPTION
**Pull request by casperisfine**
Before 29874cc4e220edadfdf50ba93db57d2df396e395 the order values
would be deduplicated when the SQL query is generated, but now
it has to be eager.

This is a minor regression we spotted when doing our weekly Rails upgrade.

cc @kamipo @rafaelfranca @Edouard-chin

### CODE REVIEW

**Comment by KapilSachdev** [comment]
Though this is a regression fix but was `Topic.reorder("id desc", id: :desc)` used to behave differently as well ?

**Comment by kamipo** [comment]
No, originally `"id desc"` and `id: :desc` cannot be deduplicated.

**Comment by KapilSachdev** [comment]
It is because the symbol argument will become an `arel` node and thus we are not inclined towards changing this as well (in case feasible)?

**Comment by kamipo** [comment]
> It is because the symbol argument will become an `arel` node

It is true.

> thus we are not inclined towards changing this as well (in case feasible)?

I don't get this part. Could you rephrase?

**Comment by KapilSachdev** [comment]
`Topic.reorder("id desc", "id desc")` will now make `ORDER BY id desc, id desc` to become `ORDER BY id desc`.

But `Topic.reorder("id desc", id: :desc)` will fetch using `ORDER BY id desc, "topics"."id" desc`.
So i meant to ask if this will continue to be the case or we would want this to change as well.

But now looking closely, `id` can be anything, so we might not be changing this behaviour.

**Comment by casperisfine** [comment]
> Topic.reorder("id desc", "id desc") will now make ORDER BY id desc, id desc to become ORDER BY id desc.

Note that this was the case previously. 29874cc changed that behavior, and this PR simply restore it.

> But Topic.reorder("id desc", id: :desc) will fetch using ORDER BY id desc, "topics"."id" desc.
> So i meant to ask if this will continue to be the case or we would want this to change as well.

The issue is that it is close to impossible. SQL fragments passed as strings are pretty much black boxes, so they are hard if not impossible to compare with arel nodes.

**Comment by KapilSachdev** [comment]
>The issue is that it is close to impossible. SQL fragments passed as strings are pretty much black boxes, so they are hard if not impossible to compare with arel nodes.

Yes, i too realised after rephrasing it in the later comment  that `id` can be anything and thus not feasible.
